First draft of p13n documentation on dev site

jonslaught opened the pull request:
@jprovine - I added this as a section below the other JS docs on developers.optimizely.com.

@tylerbrandt - this is copied directly from your client docs so mainly an FYI

tylerbrandt commented:
A preamble explaining how you call these new APIs (object with `type` property) would be welcome, as would a section for each API (outside of the examples section) where the `type` value is unambiguously stated. 

In addition, we should document which APIs are called automatically as a result of which legacy APIs (e.g. trackEvent calls `event`).

jonslaught commented:
Thx for the feedback @tylerbrandt , I added an intro about type and also added it explicitly to the parameters section and pulled that up in each section.

I decided to stay vague about which APIs are called automatically, mainly because I think we need to talk more about what our recommendation is to new customers. Should they instrument their events in both formats, or just the new format? If the new format, will we also reverse-translate for the legacy client? Will we deprecate the old format at some point?

I don't think we need to answer those questions to get this first draft onto the site for the beta customers. @jprovine let me know if you have any concerns with pushing this, otherwise I'll put it up in the next few days.
